lib/puppet/provider/sysctl/augeas.rb

```diff
@@ -39,52 +39,73 @@ def self.sysctl_get(key)
 
-  def self.instances(reference_resource = nil)
+  def self.instances(reference_resources = nil)
     return @resource_cache if @resource_cache
 
-    resources = nil
+    resources = []
+    reference_resource_titles = reference_resources.map { |_ref_name, ref_obj| ref_obj.title }
+
+    sysctl_output = ''
+    if reference_resources
+      resource_dup = reference_resources.first.last.dup
+      files = ['/etc/sysctl.conf', resource_dup[:target]].compact
+
+      files.each do |file|
+        resource_dup[:target] = file
+        augopen(resource_dup) do |aug|
+          resources ||= []
+
+          reference_resource_titles.each do |ref_title|
+	    value = aug.get("$target/#{ref_title}")
+
+	    if value
+              resource = {
+                :name    => ref_title,
+                :ensure  => :present,
+                :persist => :true,
+	        :value   => value
+              }
+
+              # Only match comments immediately before the entry and prefixed with
+              # the sysctl name
+              cmtnode = aug.match("$target/#comment[following-sibling::*[1][self::#{ref_title}]]")
+              unless cmtnode.empty?
+                comment = aug.get(cmtnode[0])
+                if comment.match(/#{resource[:name]}:/)
+                  resource[:comment] = comment.sub(/^#{resource[:name]}:\s*/, "")
+                end
+              end
+
+              resources << resource
+            end
+	  end
+        end
+      end
 
-    augopen(reference_resource) do |aug|
-      resources ||= []
+      sysctl_args = ['-e']
 
-      aug.match("$target/*").each do |spath|
-        resource = {
-          :ensure  => :present,
-          :persist => :true
-        }
+      if Facter.value(:kernel) == 'FreeBSD'
+        sysctl_args = '-ieW'
+      end
 
-        basename = spath.split("/")[-1]
-        resource[:name] = basename.split("[")[0]
-        next unless resource[:name]
-        next if (resource[:name] == "#comment")
-
-        resource[:value] = aug.get("#{spath}")
-
-        # Only match comments immediately before the entry and prefixed with
-        # the sysctl name
-        cmtnode = aug.match("$target/#comment[following-sibling::*[1][self::#{basename}]]")
-        unless cmtnode.empty?
-          comment = aug.get(cmtnode[0])
-          if comment.match(/#{resource[:name]}:/)
-            resource[:comment] = comment.sub(/^#{resource[:name]}:\s*/, "")
-          end
-        end
+      # Split this into chunks so that we don't exceed command line limits
+      reference_resource_titles.each_slice(30) do |resource_title_slice|
+        sysctl_args << resource_title_slice
 
-        resources << resource
+        sysctl_output += sysctl(sysctl_args.flatten)
       end
-    end
-
-    # Grab everything else
-    resources ||= []
+    else
+      sysctl_args = '-a'
 
-    sysctl_all_args = '-a'
-    sep = '='
+      if Facter.value(:kernel) == 'FreeBSD'
+        sysctl__args = '-aeW'
```

Comment:
Not really familar with the code, but you have a double `_` in your variable name here which was probably not intended

Reply:
@GabrielNagy Thanks, I missed that. Now fixed!